### ISSUE TITLE
Union on interfaces bypasses typecheck of property

### ISSUE THREAD

**Issue by akloeber** [Microsoft/TypeScript#29198]
**TypeScript Version:**  Version 3.3.0-dev.20181229

**Search Terms:**
- union
- interface
- typecheck

**Code**

```ts
interface Foo {
  foo: 'FOO';
}

interface Bar {
  bar: 'BAR';
}

const o: Foo | Bar = {foo: 'FOO', bar: 'xxx'};
```

**Expected behavior:**
The assignment should raise an error as the value `'xxx'` for property `bar` does not match interface `Bar`.

**Actual behavior:**
Compilation succeeds.

**Comment by ajafff** [comment]
Duplicate of #20863

**Comment by akloeber** [comment]
@ajafff Yes, looks like a duplicate of #20863. Unfortunately I didn't find this issue although I spent some time looking through issues that sounded similar.
